src/capabilities.js

```diff
@@ -198,7 +198,37 @@ class Capabilities {
 	 * @returns {Array.<FederationBackend>} Array of backends
 	 */
 	listFederation() {
-		return Array.isArray(this.data.federation) ? this.data.federation : [];
+		let federation = [];
+		if (Utils.isObject(this.data.federation)) {
+			// convert to array and add keys as `id` property
+			for(const [key, backend] of Object.entries(this.data.federation)) {
+				// fresh object to avoid `id` showing up in this.data.federation
+				federation.push({ id: key, ...backend });
+			}
+		}
+		return federation;
+	}
+
+	/**
+	 * Given just the string ID of a backend within the federation, returns that backend's full details as a FederationBackend object
+	 * 
+	 * @param {string} backendId The ID of a backend within the federation
+	 * @returns {FederationBackend | undefined} The full details of the backend, or `undefined` if no backend with the given ID exists
+	 */
+	getFederationBackendById(backendId) {
+		// Add `id` property to make it a proper FederationBackend object, but check for undefined case beforehand
+		return this.data.federation[backendId] ? { id: backendId, ...this.data.federation[backendId] } : undefined;
+	}
+
+	/**
+	 * Given a list of string IDs of backends within the federation, returns those backends' full details as FederationBackend objects
+	 * 
+	 * @param {Array<string>} backendIds The IDs of backends within the federation
```

Comment:
```suggestion
	 * @param {Array.<string>} backendIds - The IDs of backends within the federation
```
I think the missing dash is what the CI complains about, might also be the dot in the datatype.

Reply:
Indeed those are the warnings, but I think the tests eventually _fail_ because of several "AxiosError: Request failed with status code 404"...

Reply:
Yeah, but we should also get rid of the warnings if we can ;-)

![{872C58DA-495D-4238-841A-09DAA74FF02F}](https://github.com/user-attachments/assets/a31f7cd7-0d6d-4579-b50f-d41ff0a56d46)

Reply:
We definitely should! Sorry, I never do these checks locally and always only see it once they are committed and go through the CI here on Github...

So what _is_ that regex pattern??

Reply:
Not sure, but also not so imporant. I'm pretty sure we can fix it by adding the dash and the dot.

Reply:
To fix the ones from your screenshot, we needed a dot at the end of the main description.....

The others I already fixed in 303d3a6